### PR TITLE
2017080100 release candidate.

### PR DESCRIPTION
**Pull request by jmalmsten-panopto**
- Moved the synchronization timing of user enrollments between Panopto and Moodle from sign-in time to when the Panopto block is showing the contents of Panopto folder.

- Removed 'Asynchronous enrollment sync' toggle. Synchronization always runs in the block during the 'Fetching Panopto Content...' step which does not block the response on main UI.

- Removed 'Grant creator permissions to Moodle admins for all Panopto course folders' toggle, this option was prepared as the mitigation to a slow Panopto API and current block logic does not use it any more. If this behavior is still desired it is instead preferred that you go into Panopto and make the user an Panopto’s administrator from there.

- Fixed the problem when upgrading Panopto block through the command line, please see the Panopto Moodle block help page for more information on this workflow.

- Added the username of the upgrading user if the upgrade failed because of a permissions check. This should allow CLI upgrades to know who to give proper permissions to.

- Added a new check to prevent blocks from taking too long to load if the Panopto server is not available.

### CODE REVIEW

File: classes/rollingsync.php [hunk 1]
```diff
@@ -107,7 +61,7 @@ public static function coursecreated(\core\event\course_created $event) {
             $selectedserver = get_config('block_panopto', 'server_name1');
             $selectedkey = get_config('block_panopto', 'application_key1');
 
-            $task = new \block_panopto\task\provision_course();
+        $task = new \block_panopto\task\provision_course();
```
Comment (knorman-panopto) [inline classes/rollingsync.php]: Did the indentation here change for a reason?

Comment (jmalmsten-panopto) [inline classes/rollingsync.php]: Nope, I'll fix that up.

**Comment by knorman-panopto** [comment]
My only other comment on this is that it looks good to me, I don't see anything that stands out to me as incorrect. I'm not super familiar with php or the Moodle code but it looks like this accomplishes exactly what it means to do according to your description above.

**Comment by jmalmsten-panopto** [comment]
Thanks, I'll close this one up. Then open a new one with that tab fixed and merge it!

**Pull request closed without merging**